src/main/java/com/griefprevention/commands/ClaimCommand.java

```diff
@@ -105,8 +106,26 @@ else if (playerData.getClaims().isEmpty() && plugin.config_claims_automaticClaim
 
         if (radius < 0) radius = 0;
 
-        Location lesser = player.getLocation().add(-radius, 0, -radius);
-        Location greater = player.getLocation().add(radius, 0, radius);
+        Location playerLoc = player.getLocation();
+        int lesserX;
+        int lesserZ;
+        int greaterX;
+        int greaterZ;
+        try
+        {
+            lesserX = Math.subtractExact(playerLoc.getBlockX(), radius);
+            lesserZ = Math.subtractExact(playerLoc.getBlockZ(), radius);
+            greaterX = Math.addExact(playerLoc.getBlockX(), radius);
+            greaterZ = Math.addExact(playerLoc.getBlockZ(), radius);
+        }
+        catch (ArithmeticException e)
+        {
+            GriefPrevention.sendMessage(player, TextMode.Err, Messages.CreateClaimInsufficientBlocks, String.valueOf(Integer.MAX_VALUE));
+            return true;
+        }
+
+        Location lesser = new Location(world, lesserX, playerLoc.getY(), lesserZ);
+        Location greater = new Location(world, greaterX, world.getMaxHeight(), greaterZ);
```

Comment:
Is there a possibility the player could be flying above max height (e.g. creative or spectator mode)? Also interesting that bigscary uses 0 initially(?)

Reply:
That is possible, but should be corrected by claim construction if the world max height is below the player's location. In general maybe that should be treated like claiming outside the world border, but that would probably be more annoying for admins than it has value in preventing abuse by regular users elytra boosting up or something.

BigScary used 0 as a modifier to the player's current Y, not for the initial Y. Theoretically the change to making the max height the world height actually "fixes" claiming above the world by clamping the bottom of the claim to at maximum the top of the world - the claim constructor should correct incorrect min/maxes.

Reply:
Oh I see he used add.